[REF]partner_credit_limit: HU#1038 TASK#4960 CA#1, 2, 3, 4, 6, 7, 8, 9

hugho-ad opened the pull request:
- [x] Add more details in warning message when confirm a SO for a partner with overdue credit
- [x] Add page in sale order form to show credit details
- [x] Show the credit limit of the partner in Credit Details page
- [x] Show credit available of the partner in Credit Details page
- [x] Show the credit requested (sales order amount)
- [x] Show overdue amount in Credit Details page
- [x] Show overdue payments in Credit Details page 
- [x] Hide the SO's pickings when de SO state is 'credit_exeption'
- [x] Payment terms configured in main company.
- [x] add a many2many field Payment terms in order to register all payment terms "allowed  for the partner"
- [x] Add a sequence field in payment term model
- [x] Configure grace payment days in payment term model instead of res.partner
- [x] Add a group to Credit Details page.

![image](https://cloud.githubusercontent.com/assets/10233975/14287879/32a95810-fb1a-11e5-830c-53dd741554de.png)

**Dummy https://github.com/Vauxoo/yoytec/pull/921**

hugho-ad committed:
[REF]set more details in message warning when confirm a SO

hugho-ad committed:
[ADD]added new page in Sale order form to show Details credit

hugho-ad committed:
[IMP]added field many2many in partner form, in order to set the allowed payment terms

hugho-ad committed:
[REF]create a message_post instead of a warning message in SO when so pass to state 'Exception credit'

hugho-ad committed:
[REF]Hide stock.pickings when SO is in another state of 'to_invoice'

hugho-ad committed:
[FIX]set a default for new field sequence in account.payment.term

hugho-ad committed:
[REF] remove inherti function that modify domain in order to hide  the pickings when the SO is in state credit_exception

hugho-ad committed:
[FIX] fixex tests and payment_term sequence declaration

hugho-ad committed:
[REF] set payment_terms_ids in res_company

hugho-ad commented:
@gquilarque Can  you review this pr please
@JulioSerna can you help me to review merge please

Regards

JulioSerna commented on partner_credit_limit/model/sale.py:
```diff
@@ -8,27 +8,63 @@
+        for so in self:
+            moveline_obj = self.env['account.move.line']
+            movelines = moveline_obj.search(
+                [('partner_id', '=', so.partner_id.id),
```
@hugho-ad 
This is not validating contacts you must use the `comercial_partner` field

hugho-ad replied:
You men use `partner_invoice_id` field instead of `partner_id` field ?

JulioSerna replied:
This [field](https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/res/res_partner.py#L308)

JulioSerna commented on partner_credit_limit/model/sale.py:
```diff
@@ -8,27 +8,63 @@
+            movelines = moveline_obj.search(
+                [('partner_id', '=', so.partner_id.id),
+                 ('account_id.type', '=', 'receivable'),
+                 ('state', '!=', 'draft'), ('reconcile_id', '=', False)])
```
what about move line with partial reconcile, should be take into account?

hugho-ad replied:
Yes, move lines with partial reconcile are taken since, this lines also doesn't have reconcile_id, and doen't matter if there a partial reconcile, also has a overdue payment.

JulioSerna commented on partner_credit_limit/model/sale.py:
```diff
@@ -8,27 +8,63 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    partner_credit_limit = fields.Float(related='partner_id.credit_limit',
```
I think looks better in this way, like the other fields   

```
 partner_credit_limit = fields.Float(
        related='partner_id.credit_limit',
        string="Partner Credit Limit",
        readonly=True)
```

JulioSerna commented on partner_credit_limit/model/payment_term.py:
```diff
@@ -0,0 +1,23 @@
+
+    sequence = fields.Integer(
+        'Sequence', required=True,
+        default=lambda self: self.env['ir.sequence'].get(
```
`sequence` is special field of Odoo why you define one default with `ir.sequence` model?

You can see in this field the behavior of sequence,
https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/ir/ir_mail_server.py#L177

hugho-ad replied:
I took the next example , so I guess is also correct
http://toolkt.com/site/creating-a-sequence-in-odoo/

hugho-ad commented:
@JulioSerna changes Done

hugho-ad committed:
[FIX]fixed search criteria, and add amount_residual to tree view in SO

hugho-ad commented:
@JulioSerna the aml domain was corrected

Regards

hugho-ad commented:
@JulioSerna this is ready to merge

JulioSerna commented:
:+1:

hugho-ad commented:
@gquilarque can you merge this please

hugho-ad committed:
[REF] grace_payment_days field moved to account.payment.term.
The field of the number of grace days to make a payment were move to
account.payment.term and in res.partner model was created a realated
field in order to avoid the overwriting of code.
Domain wass added in sale order form for the payment_term field. For
this feature was created a realated field to res.partner
allowed payment_terms_ids.

hugho-ad committed:
[REF]added group to page of credit details

hugho-ad commented:
@gquilarque this is ready to merge
Regards

gquilarque commented:
+1 Functional Test

hugho-ad commented:
@nhomar can you help me to merge this please
Regards

hugho-ad commented:
@moylop260 can you help me to merge this please
Regards
